utils/timeout.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "timeout.h"
+
+static void timeout_handler(int signum) {
+  if (signum == SIGALRM) {
+    pthread_cancel(ta_generate_address_thread_id);
+  }
+}
+
+static void setup_timeout() {
+  signal(SIGALRM, timeout_handler);
+  setitimer(ITIMER_REAL, &TA_TIMER_TIMEOUT, NULL);
+}
+
+static void remove_timeout() {
+  signal(SIGALRM, SIG_DFL);
+  setitimer(ITIMER_REAL, &TA_ZERO_TIMEOUT, NULL);
+}
+
+static status_t ta_generate_address_thread(void *args) {
+  ta_generate_address_args_t *arg = (ta_generate_address_args_t *)args;
+  return ta_generate_address(arg->iconf, arg->service, arg->res);
+}
+
+status_t ta_generate_address_helper(const iota_config_t *const iconf, const iota_client_service_t *const service,
```

Comment:
Why did you implement `ta_generate_address_helper` inside file `timeout.c`? It should be placed into corresponding files.